### PR TITLE
On localhost launch all .NET + python examples + assistants in Aspire

### PR DESCRIPTION
**Pull request by dluc**
* When running Aspire locally, load all .NET agent examples, all python examples, all python assistants
* Fix Aspire `uv` extension not to change `uv.lock` files
* Automatically open Aspire dashboard in browser
* Improve connector ping logic

image

image

### CODE REVIEW

File: aspire-orchestrator/Aspire.Extensions/UvAppHostingExtensions.cs [hunk 1]
```diff
@@ -43,8 +43,8 @@ private static IResourceBuilder<UvAppResource> AddUvApp(this IDistributedApplica
         // var projectExecutable = instrumentationExecutable ?? pythonExecutable;
 
         string[] allArgs = args is { Length: > 0 }
-            ? ["run", scriptPath, .. args]
-            : ["run", scriptPath];
+            ? ["run", "--frozen", scriptPath, .. args]
+            : ["run", "--frozen", scriptPath];
```
Comment (dluc) [inline aspire-orchestrator/Aspire.Extensions/UvAppHostingExtensions.cs:48]: @tommasodotNET FYI the `--frozen` param cannot be after `scriptPath` so I hardcoded it here

File: libraries/dotnet/WorkbenchConnector/WorkbenchConnector.cs [hunk 2]
```diff
@@ -439,8 +439,12 @@ public virtual void Init()
                 return;
             }
 
-            // Example: http://[::]:64351
-            string first = feat.Addresses.First().Replace("[::]", "host", StringComparison.OrdinalIgnoreCase);
+            // Example: http://[::]:64351 - Prefer non-HTTPS to avoid cert validation errors
+            string first = feat.Addresses.Any(x => !x.Contains("https:", StringComparison.OrdinalIgnoreCase))
+                ? feat.Addresses.First(x => !x.Contains("https:", StringComparison.OrdinalIgnoreCase)).Replace("[::]", "host", StringComparison.OrdinalIgnoreCase)
+                : feat.Addresses.First().Replace("[::]", "host", StringComparison.OrdinalIgnoreCase);
```
Comment (dluc) [inline libraries/dotnet/WorkbenchConnector/WorkbenchConnector.cs:445]: @tommasodotNET this should fix the issue with port 5001 - e.g. if you're running an agent with 5000 (http) and 5001 (https), the code will select 5000.

File: aspire-orchestrator/Aspire.AppHost/Program.cs [hunk 3]
```diff
@@ -6,34 +6,111 @@
...
+{
+    return builder
+        .AddAssistantApp(
+            name: "skill-assistant",
+            projectDirectory: Path.Combine("..", "..", "assistants", "skill-assistant"),
+            assistantModuleName: "skill-assistant")
+        .WithEnvironment(name: "assistant__workbench_service_url", workbenchServiceEndpoint);
+}
+
+// Python Agent example: Multi-model chatbot
```
Comment (dluc) [inline aspire-orchestrator/Aspire.AppHost/Program.cs:107]: @tommasodotNET I think we need some work in the UV extension to allow loading multiple Python agents. There seems to be some problem with `assistantModuleName` and port `3001` hard coded in `AddAssistantApp`.

**Review by dluc — COMMENTED** [review]
@tommasodotNET I think we can merge this and separately address the multi-python-assistants support

File: aspire-orchestrator/Aspire.AppHost/Program.cs (hunk 3 above)
Comment (tommasodotNET) [reply aspire-orchestrator/Aspire.AppHost/Program.cs:107]: Current implementation should already allow that:
```csharp
    public static IResourceBuilder<ExecutableResource> AddAssistantApp(
        this IDistributedApplicationBuilder builder,
        string name,
        string projectDirectory,
        string assistantModuleName)
    {
        ArgumentNullException.ThrowIfNull(builder);

        var assistant = builder.AddUvApp(name, projectDirectory, "start-assistant")
            .PublishAsDockerImage(dockerContext: Path.Combine("..", ".."),
                dockerFilePath: Path.Combine("tools", "docker", "Dockerfile.assistant"),
                configure: new(configure => configure
                    .WithBuildArg("package", assistantModuleName)
                    .WithBuildArg("app", $"assistant.{assistantModuleName.Replace('-', '_')}:app")
                ));
        if (builder.ExecutionContext.IsPublishMode)
        {
            assistant.WithHttpEndpoint(port: 3001, env: "ASSISTANT__PORT");
        }
        else
        {
            assistant.WithHttpEndpoint(env: "ASSISTANT__PORT");
        }
...
```
Currently, the 3001 is hardcoded only in publish mode, when the containers ports won't get into conflict. Locally, a random port should be assigned to each assistant by Aspire using the env var ASSISTANT__PORT.

File: aspire-orchestrator/Aspire.Extensions/UvAppHostingExtensions.cs (hunk 1 above)
Comment (tommasodotNET) [reply aspire-orchestrator/Aspire.Extensions/UvAppHostingExtensions.cs:48]: We can leave it hard coded or, if it makes sense to have it chosen programmatically, we could pass a flag to this method and hard code the `--frozen` param only if the flag is true.

File: aspire-orchestrator/Aspire.AppHost/Program.cs (hunk 3 above)
Comment (dluc) [reply aspire-orchestrator/Aspire.AppHost/Program.cs:107]: @tommasodotNET Could you try uncommenting the block below? I can't find the exact param values to pass to start multi-model chatbot

Comment (dluc) [reply aspire-orchestrator/Aspire.AppHost/Program.cs]: found the problem, the path was incorrect

**Comment by dluc** [comment]
How to extend the list:

image

**Review by tommasodotNET — COMMENTED** [review]
lgtm

**Pull request merged**